velox/functions/lib/window/tests/WindowTestBase.cpp

```diff
@@ -48,6 +48,78 @@ WindowTestBase::QueryInfo WindowTestBase::buildWindowQuery(
   return {op, functionSql, querySql};
 }
 
+void WindowTestBase::buildOrderByParameters(
+    const std::string& overClause,
+    vector_size_t startIdx,
+    vector_size_t length,
+    std::vector<std::string>& orderByParsVec) {
+  auto orderByPars = overClause.substr(startIdx, length);
+  std::istringstream tokenStream(orderByPars);
+  std::string token;
+  while (std::getline(tokenStream, token, ',')) {
+    orderByParsVec.push_back(token);
+  }
+}
+
+WindowTestBase::QueryInfo WindowTestBase::buildStreamingWindowQuery(
+    const std::vector<RowVectorPtr>& input,
+    const std::string& function,
+    const std::string& overClause,
+    const std::string& frameClause) {
+  std::string functionSql =
+      fmt::format("{} over ({} {})", function, overClause, frameClause);
+  core::PlanNodePtr op = nullptr;
+  std::vector<std::string> orderByParsVec;
+
+  std::string partitionByStr = "partition by";
+  auto partitionByStartIdx = overClause.find(partitionByStr);
+
+  std::string orderByStr = "order by";
+  auto orderByStartIdx = overClause.find(orderByStr);
+
+  // Extract the partition by keys.
+  if (partitionByStartIdx != std::string::npos) {
+    auto startIdx = partitionByStartIdx + partitionByStr.length();
+    auto length = 0;
+    if (orderByStartIdx != std::string::npos) {
+      length = orderByStartIdx - startIdx;
+    } else {
+      length = overClause.length() - startIdx;
+    }
+    buildOrderByParameters(overClause, startIdx, length, orderByParsVec);
+  }
+  // Add NULLS FIRST after partition by keys. Because the
+  // defaultPartitionSortOrder in Window.cpp is NULLS FIRST.
+  for (auto i = 0; i < orderByParsVec.size(); i++) {
+    if (orderByParsVec[i].find("NULLS") == std::string::npos &&
+        orderByParsVec[i].find("nulls") == std::string::npos) {
+      orderByParsVec[i] = orderByParsVec[i] + " NULLS FIRST";
+    }
+  }
+
+  // Extract the partition by keys.
+  if (orderByStartIdx != std::string::npos) {
```

Comment:
Shouldn't the partition by keys appear before the order by keys in the ordering ?

Reply:
Yes. Updated.